Webflux

zdoh opened the pull request:
change to webFlux

zdoh committed:
update:
add generic RepositoryAggregation

fix:
ui - random kanji with no anchor show properly

zdoh committed:
v0.1.3
update:
change repository from MongoRepository to ReactiveMongoRepository

zdoh committed:
v0.1.4
fix:
application.yum fix host

zdoh committed:
v0.1.5
fix:
fix tests

zdoh committed:
v0.1.6
fix:
fix to work index.html as index file

stvort commented on src/test/java/ru/zdoher/japs/rest/WordControllerTest.java:
```diff
@@ -27,36 +34,53 @@
 
 @DisplayName("Class WordController")
 @RunWith(SpringRunner.class)
```
Это аннотация из JUnit 4. У вас используется 5й

stvort commented on src/test/java/ru/zdoher/japs/rest/WordControllerTest.java:
```diff
@@ -27,36 +34,53 @@
 @DisplayName("Class WordController")
 @RunWith(SpringRunner.class)
-@WebMvcTest(WordController.class)
+@SpringBootTest
```
Лучше `@WebMvcTest`

stvort commented on src/test/java/ru/zdoher/japs/rest/WordControllerTest.java:
```diff
@@ -27,36 +34,53 @@
+        );
+
+        WebTestClient
+                .bindToRouterFunction(function)
```
Получается вы протестировали функцию, которую создали тут же в тесте. Ничего общего с реальным контроллером она не имеет.  Лучше внедрить в тест контроллер и сделать `bindToController`

zdoh replied:
fix

zdoh committed:
v0.1.7
update:
change routing "index.html as index file" to RouterFunction in ApplicactionConfig

zdoh committed:
v0.1.8
fix:
fix controller's test

zdoh committed:
v0.1.9
fix:
change WordControllerTest from SpringBootTest to WebFluxTest

zdoh committed:
v0.1.10
update:
add some other test to WordController
